.github/workflows/samples-tools/pkg/utils/git.go

```diff
@@ -0,0 +1,15 @@
+package utils
+
+import (
+	"os/exec"
+	"strings"
+)
+
+func Diffs(head string, main string) ([]string, error) {
+	cmd := exec.Command("git", "--no-pager", "diff", "--name-only", head, main)
+	output, err := cmd.Output()
+	if err != nil {
+		return []string{}, err
```

Comment:
issue(non-blocking): wrapped CLI commands often lack critical context to debug errors, even with good messages. Even on successful output, the fact that these commands are run becomes invisible really easily.

suggestion: log the command that's run along with the working directory. Here's an example showing what I'm talking about, with some additional plumbing around ideas of labels that we probably do not need here: https://github.com/GoogleCloudPlatform/golang-samples/blob/tree/internal/cloudrunci/gcloud.go#L84

Reply:
I just moved the git command as a separate step in the workflow. It saves the output into a file and that gets passed as an argument to the `affected` command.